tlsutil/config.go

```diff
@@ -44,31 +45,62 @@ var tlsLookup = map[string]uint16{
 	"tls13": tls.VersionTLS13,
 }
 
-// Config used to create tls.Config
-type Config struct {
+// ListenerConfig contains configuration for a given listener.
```

Comment:
Definitely up for changing this! My only hesitation with "protocol" is that it makes me think of HTTP vs gRPC which is _sort of_ right except for the multiplexed port which supports multiple protocols. Maybe that's ok though, because it's more of a special case.

Reply:
Technically our multiplex port is its own protocol with extra binary data in the stream. Its sort of like how HTTP is a protocol layered on TCP which is layered on IP etc. We have gRPC layered on the "Consul multiplexer protocol".

Reply:
That's a great point! Ok, `ProtocolConfig` sounds good to me!